### PR TITLE
Espressif: fix handling of single-partition CIRCUITPY (non-CIRCUITPY_STORAGE_EXTEND) - fixes MEMENTO bug

### LINKED ISSUE #8944: Large Files on MEMENTO Causes Device Reset & Eventual Hard Crash
### CircuitPython version

```python
Adafruit CircuitPython 9.0.0-beta.1 on 2024-02-17; Adafruit Camera with ESP32S3
```

### Code/REPL

```python
# Doesn't seem to matter
print("Hello World!")
```

### Behavior

I've noticed that putting files of around 900kb in size on the MEMENTO drive seems to cause problems for the device. Originally I noticed this with a .bmp file that is 900kb I can provide that if it's though to be important, but I've since replicated the same issue using this "dummy" text file that was filled with 900kb of As, spaces and newlines. 
[dummy.txt](https://github.com/adafruit/circuitpython/files/14320692/dummy.txt)

So far I have observed the same issues when pasting the file from an Ubuntu PC and a Raspberry Pi with standard OS.

Sometimes attempting to paste the dummy.txt file once causes the device to reset and then show a hard fault message if you ctrl-c / ctrl-d in the serial console.

Other times pasting a single copy of dummy.txt suceeds, but then if you attempt to paste a second one thus far I have not seen it succeed. It has always reset part way thru the paste and then come back up with a hard fault message in serial 

Theoretically the drive should be large enough to store 3 copies of the dummy text file. But I always observe trouble from the first or second and the trouble never seems like symptoms of drive full.

When my device does get into a bad state I can recover it by deleting the large dummy file and then using `storage.erase_filesystem()` in REPL.

```
You are in safe mode because:
CircuitPython core code crashed hard. Whoops!
Hard fault: memory access or instruction error.
Please file an issue with your program at github.com/adafruit/circuitpython/issues.
Press reset to exit safe mode.
```

In cases where the dummy file was copied multiple times I renamed it to a new name before pasting the second time so it was not overwriting the original.

### Description

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by dhalbert**
- Fixes #8944

---

- Fix `supervisor_flash_write_blocks()` to write instead of read when handling a single firmware partition (thanks @jepler).
- Fix `adafruit_esp32s3_camera/mpconfigboard.mk` (MEMENTO)to use proper setting name for partition file.
- Rename partition files to have consistent names: no-ota` if only one firmware partition.
- Forbid `CIRCUITPY_DUALBANK = 1` and `CIRCUITPY_STORAGE_EXTEND = 1` if there is only one firmware partition. Thanks @jepler for initial version of validation script.

Also, some cleanups of harmless obsolete things, encountered while debugging, which are unrelated to this bug.
- Remove obsolete `*SUPERVISOR_ALLOCATION` makefile settings.
- Remove obsolete definition from `circuitpy_mpconfig.h` .

---

After loading this firmware, and doing `storage.erase_filesystem()`, the MEMENTO will have a proper CIRCUITPY filesystem (tested). Formerly it was too big.

---
While debugging this, @jepler came across this code:
https://github.com/adafruit/circuitpython/blob/499cc0158b65fe8c483ff179d99510c3f7d46812/ports/espressif/supervisor/internal_flash.c#L154-L157
We are unsure whether this code is setting `_cache_lba` to the right value.

@tannewt if you remember this code, could you take a look? But this code was present before CIRCUITPY_DUALBANK and CIRCUITPY_STORAGE_EXTEND were present.

### CODE REVIEW

File: ports/espressif/boards/adafruit_esp32s3_camera/mpconfigboard.mk [hunk 1]
```diff
@@ -12,10 +12,13 @@ CIRCUITPY_ESP_FLASH_FREQ = 80m
 CIRCUITPY_ESP_PSRAM_SIZE = 2MB
 CIRCUITPY_ESP_PSRAM_MODE = qio
 CIRCUITPY_ESP_PSRAM_FREQ = 80m
-FLASH_SDKCONFIG = esp-idf-config/sdkconfig-4MB-1ota.defaults
+
+# No OTA partition: larger filesystem instead for lots of images.
```
Comment (jepler) [inline ports/espressif/boards/adafruit_esp32s3_camera/mpconfigboard.mk:16]: isn't it designed for a larger firmware, not more filesystem?

**Review by jepler — CHANGES_REQUESTED** [review]
Thank you -- excellent cleanups done along the way. However, can you double check this comment you added? It's the opposite of what I understood from the discussions.

File: ports/espressif/boards/adafruit_esp32s3_camera/mpconfigboard.mk (hunk 1 above)
Comment (dhalbert) [reply ports/espressif/boards/adafruit_esp32s3_camera/mpconfigboard.mk:16]: yes, that is wrong, from an early misunderstanding of mine.

**Comment by FoamyGuy** [comment]
I tested a build from this branch successfully on the device. With this version the drive size reported to OS makes it so it will warn me that the file is too large to fit when I try pasting it if there is already other things on the drive.

I also have been able to successfully use this build with a modified version of the basic camera example from learn and been able to take a few photos successfully with it.

**Review by jepler — APPROVED** [review]
thanks for fixing that!

**Comment by tannewt** [comment]
> @tannewt if you remember this code, could you take a look? But this code was present before CIRCUITPY_DUALBANK and CIRCUITPY_STORAGE_EXTEND were present.

It doesn't look right to me and it looks like RP2040 has the same bug. I've always meant to factor out this cache code but never have...

**Pull request merged**